python/cuda_parallel/tests/test_segmented_reduce.py

```diff
@@ -4,31 +4,41 @@
 
 import cupy as cp
 import numpy as np
+import pytest
 
 import cuda.parallel.experimental.algorithms as algorithms
 from cuda.parallel.experimental.struct import gpu_struct
 
 
-def test_segmented_reduce(input_array):
-    "Test for all supported input types"
+@pytest.fixture(params=["i4", "u4", "i8", "u8"])
+def offset_dtype(request):
+    return np.dtype(request.param)
+
+
+def test_segmented_reduce(input_array, offset_dtype):
+    "Test for all supported input types and for some offset types"
 
     def binary_op(a, b):
         return a + b
 
     assert input_array.ndim == 1
     sz = input_array.size
     rng = np.random.default_rng()
-    n_segments = 2**4
+    n_segments = 16
     h_offsets = np.zeros(n_segments + 1, dtype="int64")
-    h_offsets[1:] = rng.multinomial(sz, [1 / 16] * 16)
+    h_offsets[1:] = rng.multinomial(sz, [1 / n_segments] * n_segments)
 
-    offsets = cp.asarray(h_offsets)
+    offsets = cp.cumsum(cp.asarray(h_offsets, dtype=offset_dtype), dtype=offset_dtype)
 
     start_offsets = offsets[:-1]
-    end_offsets = offsets[:-1]
+    end_offsets = offsets[1:]
```

Comment:
I take it that this is the actual fix.

Reply:
This definitely is a required component of the fix, but this alone won't do it, since the random draw from the multinomial distribution does not produce a non-decreasing sequence, and hence we won't be reducing all elements in the input array, and many segments will be empty, since `end_offsets` value might be less than the corresponding `start_offsets` value.

Reply:
Yes, so you added a scan to produce valid segment lengths